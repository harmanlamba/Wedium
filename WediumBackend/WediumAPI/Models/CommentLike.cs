using System;
using System.Collections.Generic;

namespace WediumAPI.Models
{
    public partial class CommentLike
    {
        public int CommentLikeId { get; set; }
        public int CommentId { get; set; }
        public int UserId { get; set; }
        public DateTime Date { get; set; }

        public virtual Comment Comment { get; set; }
        public virtual User User { get; set; }
    }
}
