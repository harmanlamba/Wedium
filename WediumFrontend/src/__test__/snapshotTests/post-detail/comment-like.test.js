import React from 'react';
import { shallow } from 'enzyme';

import CommentLike from '../../../components/post-detail/comment/comment-like';

it('snapshot commentLike default load', () => {
  const user = 1;
  const numberOfLikes = 0;
  const commentId = 1;
  const parentCommentId = null;
  const isCommentLiked = false;

  let wrapper = shallow(
    <CommentLike
      user={user}
      numberOfLikes={numberOfLikes}
      commentId={commentId}
      parentCommentId={parentCommentId}
      isCommentLiked={isCommentLiked}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('snapshot commentLike for child comment load', () => {
  const user = 1;
  const numberOfLikes = 0;
  const commentId = 2;
  const parentCommentId = 1;
  const isCommentLiked = false;

  let wrapper = shallow(
    <CommentLike
      user={user}
      numberOfLikes={numberOfLikes}
      commentId={commentId}
      parentCommentId={parentCommentId}
      isCommentLiked={isCommentLiked}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('snapshot commentLike with liked component load', () => {
  const user = 1;
  const numberOfLikes = 5;
  const commentId = 2;
  const parentCommentId = 1;
  const isCommentLiked = true;

  let wrapper = shallow(
    <CommentLike
      user={user}
      numberOfLikes={numberOfLikes}
      commentId={commentId}
      parentCommentId={parentCommentId}
      isCommentLiked={isCommentLiked}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
