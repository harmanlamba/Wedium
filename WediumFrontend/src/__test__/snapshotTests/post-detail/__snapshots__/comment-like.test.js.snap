// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot commentLike default load 1`] = `
<Connect(CommentLike)
  classes={
    Object {
      "likedIcon": "Connect(CommentLike)-likedIcon-3",
      "text": "Connect(CommentLike)-text-1",
      "unlikedIcon": "Connect(CommentLike)-unlikedIcon-2",
    }
  }
  commentId={1}
  isCommentLiked={false}
  numberOfLikes={0}
  parentCommentId={null}
  user={1}
/>
`;

exports[`snapshot commentLike for child comment load 1`] = `
<Connect(CommentLike)
  classes={
    Object {
      "likedIcon": "Connect(CommentLike)-likedIcon-3",
      "text": "Connect(CommentLike)-text-1",
      "unlikedIcon": "Connect(CommentLike)-unlikedIcon-2",
    }
  }
  commentId={2}
  isCommentLiked={false}
  numberOfLikes={0}
  parentCommentId={1}
  user={1}
/>
`;

exports[`snapshot commentLike with liked component load 1`] = `
<Connect(CommentLike)
  classes={
    Object {
      "likedIcon": "Connect(CommentLike)-likedIcon-3",
      "text": "Connect(CommentLike)-text-1",
      "unlikedIcon": "Connect(CommentLike)-unlikedIcon-2",
    }
  }
  commentId={2}
  isCommentLiked={true}
  numberOfLikes={5}
  parentCommentId={1}
  user={1}
/>
`;
